### PR TITLE
[glfw3] Add wayland feature

### LINKED ISSUE #28230: [glfw3] Add Wayland support as a Library Feature
****

**Is your feature request related to a problem? Please describe.**
When compiling glfw3 on Fedora 37, compilation fails because the default display server in Fedora is new Wayland, but by default glfw3 expects the X Window system on Linux. However, glfw3 does have a config flag which builds it with Wayland support.

**Proposed solution**
I propose the addition of `wayland` as a Library Feature so that this flag can be selected through the usual install / manifest mechanism.

**Describe alternatives you've considered**
I'm not aware of a way to work around this, certainly not in manifest mode.

**Additional context**
I've modified my local copies of `portfile.cmake` and `vcpkg.json` to implement this feature. If people think it sounds like a good idea, I'll be happy to create a pull request.

### PR DESCRIPTION
**Pull request by marinedalek**
This PR adds a `wayland` feature to the glfw3 port to enable compilation targeting the Wayland display server.

- #### What does your PR fix?
  Fixes #28230 

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
Unchanged

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
 Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by marinedalek** [comment]
@microsoft-github-policy-service agree

File: ports/glfw3/vcpkg.json [hunk 1]
```diff
@@ -14,5 +14,11 @@
       "name": "vcpkg-cmake-config",
       "host": true
     }
-  ]
+  ],
+  "features": {
+    "wayland": {
+      "description": "Use Wayland display server",
+      "supports": "linux"
+    }
```
Comment (Neumann-A) [inline ports/glfw3/vcpkg.json:22]: Please add the required port dependencies for this.

Comment (Neumann-A) [reply ports/glfw3/vcpkg.json:22]: https://github.com/microsoft/vcpkg/blob/master/ports/wayland/vcpkg.json and there is also `wayland-protocols`.

Comment (marinedalek) [reply ports/glfw3/vcpkg.json:22]: The empty ports `wayland`, `wayland-protocols`, `libxkbcommon`, and `ecm` currently build their dependencies unnecessarily, even when compilation is not forced. In addition, `pthread-stubs`, one of `libxkbcommon`'s transitive dependencies, does not build successfully on a clean install of Fedora. Upon initial failure, a diagnostic message is displayed instructing the user to install `autoconf` from the system package manager. However, a subsequent build attempt also fails with no such helpful message, requiring a dig into the logs and a web search to discover that `automake` is also required. 
Given these issues, I'd rather not add these dependencies at this time. Is that OK? As an alternative, I have modified the build message for linux builds with a link to the GLFW docs that list the required packages for Wayland compilation on a variety of distros. That change is now part of this pull request.

**Review by Adela0814 — DISMISSED** [review]

File: ports/glfw3/vcpkg.json (hunk 1 above)
Comment (vicroms) [reply ports/glfw3/vcpkg.json:22]: I have made PR #28628 to disable the extra dependencies when not actually building `wayland`. The same approach should be used for the other `X_VCPKG_[X|WAYLAND]_FORCE_VCPKG_LIBRARIES` and after that is done then we can add the dependencies on this port.

Comment (vicroms) [reply ports/glfw3/vcpkg.json:22]: #28628 has been merged so you should be able to declare the dependency on `wayland` and it won't build unnecessary dependencies anymore.

File: ports/glfw3/vcpkg.json [hunk 2]
```diff
@@ -14,5 +14,11 @@
       "name": "vcpkg-cmake-config",
       "host": true
     }
-  ]
+  ],
+  "features": {
+    "wayland": {
+      "description": "Use Wayland display server",
+      "supports": "linux"
```
Comment (vicroms) [inline ports/glfw3/vcpkg.json:21]:
```suggestion
      "supports": "linux",
      "dependencies": [ "wayland" ]
```

**Review by vicroms — CHANGES_REQUESTED** [review]

**Review by vicroms — APPROVED** [review]

**Pull request merged**